src/index.js

```diff
@@ -28,54 +23,26 @@ class DelegatedPeerRouting {
    * @param {PeerID} id
    * @param {object} options
    * @param {number} options.maxTimeout How long the query can take. Defaults to 30 seconds
-   * @param {function(Error, PeerInfo)} callback
-   * @returns {void}
+   * @returns {Promise}
```

Comment:
```suggestion
   * @returns {Promise<PeerInfo>}
```

Reply:
Me, personally, I don't like jsdocs and this is a clear example of why.

This module calls `this.dht.findPeer` which returns a Promise of... a `PeerInfo`?  Now yes, but in the future maybe not and there's nothing to let us know.

That is, we treat them like types and generate documentation from them as if they are accurate but really they are out of date as soon as they've been typed out.

Reply:
The advantage of this for me is that I can get introspection when I'm writing code, and I don't have to go look at docs or readme's. Resolves/Rejects don't have good support yet in jsdocs. I'd prefer the approach mentioned at https://github.com/jsdoc/jsdoc/issues/1467. In reality, having a return type of promise is redundant due to the function declaration of async. Knowing what's going to be resolved/rejected is the valuable bit.

Reply:
It's nice when this works but it's just too easy for them to get out of date.

In the case above it wouldn't even be the fault of this module - if `this.dht.findPeer` changed it's return type and we just blindly pass it through there's nothing (like a compiler) to tell us we are breaking our users.

The closest thing we have to automation there is someone opening an issue or sending a PR which is not ideal.